ddl/label/rule.go

```diff
@@ -85,10 +85,65 @@ func (r *Rule) Clone() *Rule {
 // ResetTable will reset the label rule for a table with a given ID and names.
 func (r *Rule) ResetTable(id int64, dbName, tableName string) *Rule {
 	r.ID = fmt.Sprintf(TableIDFormat, IDPrefix, dbName, tableName)
-	r.Labels = append(r.Labels, []Label{
-		{Key: dbKey, Value: dbName},
-		{Key: tableKey, Value: tableName},
-	}...)
+
+	var hasDBKey, hasTableKey bool
+	for _, label := range r.Labels {
+		if label.Key == dbKey {
+			label.Value = dbName
+			hasDBKey = true
+		}
+		if label.Key == tableKey {
+			label.Value = tableName
+			hasTableKey = true
+		}
+	}
+
+	if !hasDBKey {
+		r.Labels = append(r.Labels, Label{Key: dbKey, Value: dbName})
+	}
+
+	if !hasTableKey {
+		r.Labels = append(r.Labels, Label{Key: tableKey, Value: tableName})
+	}
+
+	r.RuleType = ruleType
+	r.Rule = map[string]string{
+		"start_key": hex.EncodeToString(codec.EncodeBytes(nil, tablecodec.GenTableRecordPrefix(id))),
+		"end_key":   hex.EncodeToString(codec.EncodeBytes(nil, tablecodec.GenTableRecordPrefix(id+1))),
+	}
+	return r
+}
+
+// ResetPartition will reset the label rule for a partition with a given ID and names.
+func (r *Rule) ResetPartition(id int64, dbName, tableName, partName string) *Rule {
```

Comment:
Seems repeative with `ResetTable`, any idea on how to liteweight/reuse the code?